alvr/server_openvr/src/props.rs

```diff
@@ -654,53 +477,94 @@ pub extern "C" fn set_device_openvr_props(device_id: u64) {
+            }
 
-                    // icons
-                    set_prop(
-                        NamedIconPathDeviceOffString,
-                        "{htc}/icons/tracker_status_off.png",
-                    );
-                    set_prop(
-                        NamedIconPathDeviceSearchingString,
-                        "{htc}/icons/tracker_status_searching.gif",
-                    );
-                    set_prop(
-                        NamedIconPathDeviceSearchingAlertString,
-                        "{htc}/icons/tracker_status_searching_alert.gif",
-                    );
-                    set_prop(
-                        NamedIconPathDeviceReadyString,
-                        "{htc}/icons/tracker_status_ready.png",
-                    );
-                    set_prop(
-                        NamedIconPathDeviceReadyAlertString,
-                        "{htc}/icons/tracker_status_ready_alert.png",
-                    );
-                    set_prop(
-                        NamedIconPathDeviceNotReadyString,
-                        "{htc}/icons/tracker_status_error.png",
-                    );
+            // Match for icons, to avoid duplicating across every Meta headset
+            // and also to override for the hand tracking controllers specifically
+            match config.emulation_mode {
+                ControllersEmulationMode::RiftSTouch
+                | ControllersEmulationMode::Quest2Touch
+                | ControllersEmulationMode::Quest3Plus
+                | ControllersEmulationMode::QuestPro => {
+                    set_prop(TrackingSystemNameString, "oculus");
+
+                    set_prop(ControllerTypeString, "oculus_touch");
+                    set_prop(InputProfilePathString, "{oculus}/input/touch_profile.json");
+                    if left_hand {
+                        set_prop(
+                            RegisteredDeviceTypeString,
+                            format!("oculus/{headset_serial}_Controller_Left").as_str(),
+                        );
+                        set_icons("{oculus}/icons/rifts_left_controller");
+                    } else if right_hand {
+                        set_prop(
+                            RegisteredDeviceTypeString,
+                            format!("oculus/{headset_serial}_Controller_Right").as_str(),
+                        );
+                        set_icons("{oculus}/icons/rifts_right_controller");
+                    }
+                }
+                ControllersEmulationMode::ValveIndex => {
+                    if left_hand {
+                        set_icons("{indexcontroller}/icons/left_controller_status");
+                    } else if right_hand {
+                        set_icons("{indexcontroller}/icons/right_controller_status");
+                    }
+                }
+                // This is probably also a good switch statement as a fallback
```

Comment:
Move this code back to the previous match and reduplicate some entries if necessary. It's not that I prefer duplicate entries but it makes things cleaner overall.

Reply:
I'll compromise and make it a closure instead

Reply:
Yes you can make a reusable closure, but merge the match blocks.

Reply:
oh I already did that, check the new diff